test/e2e/suite/conformance/certificatesigningrequests/suite.go

```diff
@@ -73,6 +77,17 @@ type Suite struct {
 	// nginx-ingress addon.
 	DomainSuffix string
 
+	// IPAddress is an IP used on all IPAddress requests.
+	// This is useful when the issuer being tested requires special
+	// configuration for a set of IPAddresses in order for certificates to be
+	// issued, such as the ACME issuer.
+	// If not set, this will be defaulted to "127.0.0.1".
+	IPAddress string
+
+	// IPAddressType is a string used to detect what IPAddress to use
+	// in case IPAddress is left empty.
+	IPAddressType string
```

Comment:
Is this required? It would seem to me to just have `IPAdress` defined, which will be defaulted if empty.

`f.Config.Addons.ACMEServer.IngressIP` should be able to be referenced by issuers that need to set it.

Reply:
I think the problem was that in eg. `runACMEIssuerTests` you do not have access to `f.Config.Addons.ACMEServer.IngressIP` yet so you cannot override the `IPAddress` value with that value. The `f *framework.Framework` parameter is only injected at the moment the `complete()` function is called

Reply:
I see, then maybe better to change this to `UseIngressIPAddress bool` in the suite struct, and remove both `IPAddress` and `IPAddressType`.

We can then define a `sharedIPAddress` in `tests.go`, similar to [`sharedURI`](https://github.com/jetstack/cert-manager/blob/15b6f32071b45687b44eb097145815fc3af1b010/test/e2e/suite/conformance/certificatesigningrequests/tests.go#L61), which implements the logic to determine which IP to use based on `UseIngressIPAddress`.